cmake/OpenCVCompilerOptions.cmake

```diff
@@ -41,14 +41,18 @@ if((CV_CLANG AND NOT CMAKE_GENERATOR MATCHES "Xcode")  # PCH has no support for
 endif()
 
 macro(add_extra_compiler_option option)
-  ocv_check_flag_support(CXX "${option}" _varname "${OPENCV_EXTRA_CXX_FLAGS} ${ARGN}")
-  if(${_varname})
-    set(OPENCV_EXTRA_CXX_FLAGS "${OPENCV_EXTRA_CXX_FLAGS} ${option}")
-  endif()
+  ocv_check_flag_support(CXX "${option}" _varname
+                         UPDATED_COMPILER_FLAGS OPENCV_EXTRA_CXX_FLAGS
+                         UPDATED_LINKER_FLAGS OPENCV_EXTRA_EXE_LINKER_FLAGS
+                         ${ARGN})
+
+  ocv_check_flag_support(C "${option}" _varname
+                         UPDATED_COMPILER_FLAGS OPENCV_EXTRA_C_FLAGS
+                         UPDATED_LINKER_FLAGS OPENCV_EXTRA_EXE_LINKER_FLAGS
+                         ${ARGN})
 
-  ocv_check_flag_support(C "${option}" _varname "${OPENCV_EXTRA_C_FLAGS} ${ARGN}")
   if(${_varname})
-    set(OPENCV_EXTRA_C_FLAGS "${OPENCV_EXTRA_C_FLAGS} ${option}")
+    ocv_list_unique(OPENCV_EXTRA_EXE_LINKER_FLAGS)
```

Comment:
I think it can break something. For example if we had flags like this: `-fenable-something -fdisable-something -fenable-something`, we can get `-fenable-something -fdisable-something` which is not equivalent.